sdk/cosmos/azure-cosmos/test/test_container_properties_cache_async.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# The MIT License (MIT)
+# Copyright (c) Microsoft Corporation. All rights reserved.
+
+"""End-to-end test.
+"""
+
+import unittest
+import uuid
+
+import pytest
+
+import azure.cosmos.exceptions as exceptions
+import test_config
+from azure.cosmos.aio import CosmosClient, _retry_utility_async, DatabaseProxy
+from azure.cosmos.partition_key import PartitionKey
+
+
+@pytest.mark.cosmosEmulator
+class TestContainerPropertiesCache(unittest.IsolatedAsyncioTestCase):
+    """Python CRUD Tests.
+        """
+
+    client: CosmosClient = None
+    configs = test_config.TestConfig
+    host = configs.host
+    masterKey = configs.masterKey
+    connectionPolicy = configs.connectionPolicy
+    database_for_test: DatabaseProxy = None
+
+    @classmethod
+    def setUpClass(cls):
+        if (cls.masterKey == '[YOUR_KEY_HERE]' or
+                cls.host == '[YOUR_ENDPOINT_HERE]'):
+            raise Exception(
+                "You must specify your Azure Cosmos account values for "
+                "'masterKey' and 'host' at the top of this class to run the "
+                "tests.")
+
+    async def asyncSetUp(self):
+        self.client = CosmosClient(self.host, self.masterKey)
+        self.database_for_test = self.client.get_database_client(self.configs.TEST_DATABASE_ID)
+
+    async def tearDown(self):
+        await self.client.close()
+
+    async def test_container_properties_cache(self):
+        client = self.client
+        database_name = "Container Properties Cache Test DB Async " + str(uuid.uuid4())
+        created_db = await client.create_database(database_name)
+        container_name = str(uuid.uuid4())
+        container_pk = "PK"
+        # Create The Container
+        try:
+            await client.get_database_client(database_name).create_container(id=container_name, partition_key=PartitionKey(
```

Comment:
any reason why you are using the database proxy here instead of the created_db from the create operation?

Reply:
Its to check the cache while using the proxies like this instead of an instance like how it is usually done, since that's the scenario in which the old cache had issues being used. I will also be expanding this tests in the followup PR.